### PR TITLE
Build and push CI/CD GitHub Action

### PR DESCRIPTION
**Pull request by Ayedrian**
Only significant change to the script is the different "runs-on", the current Dockerfile uses an optimized Debian bullseye distro for Node

### CODE REVIEW

**Review by Mtze — CHANGES_REQUESTED** [review]

File: .github/workflows/build-and-push.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+      -
+        name: Checkout
+        uses: actions/checkout@v3
+
+      -
+        name: Docker Login
+        id: docker-login
+        run: |
+          docker login -u ${{secrets.DOCKER_USERNAME}} -p ${{secrets.DOCKER_PASSWORD}}
```
[the remark this answers is not in the document]
Comment (Ayedrian) [reply .github/workflows/build-and-push.yml:41]: For the dockerhub login, are the secrets the same or are they referred to differently such as DOCKERHUB_USERNAME etc.?

[the remark this answers is not in the document]
Comment (Mtze) [reply .github/workflows/build-and-push.yml]: The secrets names are fine.

[the remark this answers is not in the document]
Comment (Ayedrian) [reply .github/workflows/build-and-push.yml]: Nice, last commit seems to have produced a package.

**Comment by Mtze** [comment]
For the future: Please use the suggestions in the PR - otherwise it is really hard and cumbersome to keep track about the changes.

File: .github/workflows/build-and-push.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+name: Build and Push
+
+on:
+  push:
```
[the remark this answers is not in the document]
Comment (Mtze) [reply .github/workflows/build-and-push.yml]: I think you missed the `push` in your current version

[the remark this answers is not in the document]
Comment (Ayedrian) [reply .github/workflows/build-and-push.yml]: Your suggested change was to remove it, no?

[the remark this answers is not in the document]
Comment (Mtze) [reply .github/workflows/build-and-push.yml]: Have a close look at my suggestion...

**Comment by Ayedrian** [comment]
> For the future: Please use the suggestions in the PR - otherwise it is really hard and cumbersome to keep track about the changes.

Is this just referring to the missing push?

**Pull request merged**